SEP-0012.md

```diff
@@ -0,0 +1,343 @@
+                                 *axes: Union[Union[int, str], Iterable[Union[int, str]]],
+                                 edges: bool = False) -> Iterable[u.Quantity]:
+        """
+        Returns world coordinate values of all pixels for all axes.
+
+        Parameters
+        ----------
+        axes
+            Axis number in numpy ordering or unique substring of
+            `.NDCube.wcs.world_axis_physical_types` or
+            `.NDCube.wcs.world_axis_names` of axes for which real world
+            coordinates are desired. ``axes=None`` implies all axes will be
+            returned.
+
+        edges
+            The edges argument helps in returning ``pixel_edges``
+            instead of ``pixel_values``. Default value is False,
+            which returns ``pixel_values``. True return ``pixel_edges``
+
+        Returns
+        -------
+        axes_coords
+            `~astropy.units.Quantity` or iterable thereof for all requested
+            world axes, units determined by the wcs.  # TODO: Should we mandate namedtuple?
+
+        Example
+        -------
+        >>> NDCube.all_world_coords_values(('lat', 'lon'))
+        >>> NDCube.all_world_coords_values(2)
+
+        """
+
+    @abc.abstractmethod
+    def crop(self,
+             lower_corner: Iterable[Union[u.Quantity, float]],
+             upper_corner: Iterable[Union[u.Quantity, float]] = None,
+             units: Iterable[Union[str, u.Unit]] = None,
+             wcs: BaseHighLevelWCS = None):
+        """
+        Crops an NDCube given minimum values and interval widths along axes.
+
+        Parameters
+        ----------
+        lower_corner
+            The minimum desired values along each relevant axis after cropping
+            described in physical units consistent with the NDCube's wcs object.
+            The length of the iterable must equal the number of data dimensions
+            and must have the same order as the data.
+
+        upper_corner
+            The maximum desired values along each relevant axis after cropping
+            described in physical units consistent with the NDCube's wcs object.
+            The length of the iterable must equal the number of data dimensions
+            and must have the same order as the data.
+
+        units
+            If the inputs are set without units, the user must set the units
+            inside this argument as `str` or `~astropy.units.Unit` objects.
+            The length of the iterable must equal the number of data dimensions
+            and must have the same order as the data.
+
+        wcs
+            The WCS to use to calculate the pixel coordinates based on the input.
+            Will default to the ``.wcs`` property if not given. While any valid WCS
+            could be used it is expected that either the ``.wcs`` or
+            ``.extra_coords.wcs`` properties will be used.
+
+        Returns
+        -------
+        result: NDCube
+        """
+    
+```
+
+## Extra Coords
```

Comment:
I know this is my own ignorance of extra_coords but can you give an example of one such way this is used in the context of for example a certain type of observation?

Reply:
Say you have a 3-D cube representing a raster scan.  The x-dimension corresponds to longitude in the WCS, but because the raster is compiled by taking an exposure at different slit position sequentially, it also corresponds to time.  If the WCS only have lat, lon, wavelength, you can create a time extra coordinate and assign it to the x-dimension.

Reply:
ah yes I see - is it worth adding this in the SEP just for readers not necessarily familiar?

Reply:
Sure.  You can suggest a change if you like via GitHub

Reply:
I totally agree with this comment and suggest a whole new section be added which provides examples use case of this API.

Reply:
I have added a small section with a high level overview of the three classes, however, if you want examples I refer you to the new NDCube docs.